modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/item/ContentDashboardItemFactoryTracker.java

```diff
@@ -67,20 +90,174 @@ protected void activate(BundleContext bundleContext) {
+				new ContentDashboardItemFactoryTrackerCustomizer(
+					bundleContext));
 	}
 
 	@Deactivate
 	protected void deactivate() {
 		_serviceTrackerMap.close();
 	}
 
+	private void _addAssetVocabulary(
+			Company company, long classNameId, String name, int visibilityType)
+		throws Exception {
+
+		User defaultUser = company.getDefaultUser();
+
+		Map<Locale, String> titleMap = new HashMap<>();
+
+		for (Locale locale :
+				_language.getCompanyAvailableLocales(company.getCompanyId())) {
+
+			titleMap.put(locale, LanguageUtil.get(locale, name));
+		}
+
+		AssetVocabularySettingsHelper assetVocabularySettingsHelper =
+			new AssetVocabularySettingsHelper();
+
+		assetVocabularySettingsHelper.setClassNameIdsAndClassTypePKs(
+			new long[] {classNameId},
+			new long[] {AssetCategoryConstants.ALL_CLASS_TYPE_PK},
+			new boolean[] {false});
+
+		ServiceContext serviceContext = new ServiceContext();
+
+		serviceContext.setAddGroupPermissions(true);
+		serviceContext.setAddGuestPermissions(true);
+
+		_assetVocabularyLocalService.addVocabulary(
+			null, defaultUser.getUserId(), company.getGroupId(), name,
+			StringPool.BLANK, titleMap, Collections.emptyMap(),
+			assetVocabularySettingsHelper.toString(), visibilityType,
+			serviceContext);
+	}
+
+	private void _addClassNameIdToVocabularyIfNotExist(
+		long classNameId, String name) {
+
+		try {
+			_companyLocalService.forEachCompany(
+				company -> {
+					AssetVocabulary assetVocabulary =
+						_assetVocabularyLocalService.fetchGroupVocabulary(
+							company.getGroupId(),
+							StringUtil.toLowerCase(GetterUtil.getString(name)));
+
+					if (assetVocabulary == null) {
+						_addAssetVocabulary(
+							company, classNameId, name,
+							AssetVocabularyConstants.VISIBILITY_TYPE_INTERNAL);
+
+						return;
+					}
+
+					AssetVocabularySettingsHelper
+						assetVocabularySettingsHelper =
+							new AssetVocabularySettingsHelper();
+
+					long[] selectedClassNameIds =
+						assetVocabulary.getSelectedClassNameIds();
+
+					if (!ArrayUtil.contains(
+							selectedClassNameIds, classNameId)) {
+
+						long[] classNameIds = ArrayUtil.append(
+							assetVocabulary.getSelectedClassNameIds(),
+							classNameId);
+						long[] classTypePKs = ArrayUtil.append(
+							assetVocabulary.getSelectedClassTypePKs(),
+							AssetCategoryConstants.ALL_CLASS_TYPE_PK);
+
+						boolean[] requireds =
+							new boolean[selectedClassNameIds.length + 1];
+
+						for (int i = 0; i < requireds.length; i++) {
+							requireds[i] = false;
+						}
+
+						assetVocabularySettingsHelper.
+							setClassNameIdsAndClassTypePKs(
+								classNameIds, classTypePKs, requireds);
+					}
+
+					_assetVocabularyLocalService.updateVocabulary(
+						assetVocabulary.getVocabularyId(),
+						assetVocabulary.getTitleMap(),
+						assetVocabulary.getDescriptionMap(),
+						assetVocabularySettingsHelper.toString(),
+						AssetVocabularyConstants.VISIBILITY_TYPE_INTERNAL);
```

Comment:
We are hardcoding the type of visibility here so maybe the method should reflect it, what do you think?
Could be a bit long something like `_addClassNameIdToInternalVocabularyIfNotExist`

Or maybe pass the type as an argument? 🤷

Reply:
Maybe the parameter approach is a good one :).

Reply:
Could you change it?

Reply:
Sure!